src/store/inflight_activation.rs

```diff
@@ -563,19 +563,15 @@ impl InflightActivationStore {
     #[instrument(skip_all)]
     pub async fn handle_expires_at(&self) -> Result<u64, Error> {
         let now = Utc::now();
-        let update_result = sqlx::query(
-            r#"UPDATE inflight_taskactivations
-            SET status = $1
-            WHERE expires_at IS NOT NULL AND expires_at < $2 AND status = $3
-            "#,
+        let query = sqlx::query(
+            "DELETE FROM inflight_taskactivations WHERE expires_at IS NOT NULL AND expires_at < $1",
```

Comment:
Shouldn't we also keep the condition on status = Pending? There is a bind for it still as well.